ginisdk/src/androidTest/java/net/gini/android/SdkIntegrationTestAccounting.java

```diff
@@ -0,0 +1,361 @@
+                setApiBaseUrl(apiUriAccounting).
+                setGiniApiType(GiniApiType.ACCOUNTING).
+                setUserCenterApiBaseUrl(userCenterUri).
+                setConnectionTimeoutInMs(60000).
+                setCredentialsStore(credentialsStore).
+                build();
+
+        processDocument(uploadBuilder);
+
+        UserCredentials newUserCredentials = credentialsStore.getUserCredentials();
+        assertEquals(newEmailDomain, extractEmailDomain(newUserCredentials.getUsername()));
+    }
+
+    public void testPublicKeyPinningWithMatchingPublicKey() throws Exception {
+        resetTrustKit();
+        gini = new SdkBuilder(getContext(), clientId, clientSecret, "example.com").
+                setNetworkSecurityConfigResId(net.gini.android.test.R.xml.network_security_config).
+                setApiBaseUrl(apiUriAccounting).
+                setGiniApiType(GiniApiType.ACCOUNTING).
+                setUserCenterApiBaseUrl(userCenterUri).
+                setConnectionTimeoutInMs(60000).
+                build();
+        final AssetManager assetManager = getContext().getResources().getAssets();
+        final InputStream testDocumentAsStream = assetManager.open("test.jpg");
+        assertNotNull("test image test.jpg could not be loaded", testDocumentAsStream);
+
+        final byte[] testDocument = TestUtils.createByteArray(testDocumentAsStream);
+        final DocumentUploadBuilder uploadBuilder = new DocumentUploadBuilder().setDocumentBytes(testDocument).setDocumentType(DocumentTaskManager.DocumentType.INVOICE);
+        processDocument(uploadBuilder);
+    }
+
+    public void testPublicKeyPinningWithCustomCache() throws Exception {
+        resetTrustKit();
+        gini = new SdkBuilder(getContext(), clientId, clientSecret, "example.com").
+                setNetworkSecurityConfigResId(net.gini.android.test.R.xml.network_security_config).
+                setApiBaseUrl(apiUriAccounting).
+                setGiniApiType(GiniApiType.ACCOUNTING).
+                setUserCenterApiBaseUrl(userCenterUri).
+                setConnectionTimeoutInMs(60000).
+                setCache(new NoCache()).
+                build();
+
+        final AssetManager assetManager = getContext().getResources().getAssets();
+        final InputStream testDocumentAsStream = assetManager.open("test.jpg");
+        assertNotNull("test image test.jpg could not be loaded", testDocumentAsStream);
+
+        final byte[] testDocument = TestUtils.createByteArray(testDocumentAsStream);
+        final DocumentUploadBuilder uploadBuilder = new DocumentUploadBuilder().setDocumentBytes(testDocument).setDocumentType(DocumentTaskManager.DocumentType.INVOICE);
+        processDocument(uploadBuilder);
+    }
+
+    @SdkSuppress(maxSdkVersion = Build.VERSION_CODES.JELLY_BEAN_MR1)
+    public void testPublicKeyPinningWithWrongPublicKey() throws Exception {
+        resetTrustKit();
```

Comment:
![](https://d2skuhm0vrry40.cloudfront.net/2015/articles/1/8/7/0/5/6/3/pokemon-go-ditto-how-to-find-and-catch-ditto-in-the-wild-and-everything-else-you-need-to-know-1479913157461.jpg/EG11/thumbnail/1920x1080/format/jpg/2009834.jpg)